hack/generator/pkg/astmodel/package_reference.go

```diff
@@ -36,3 +44,219 @@ func PackageAsLocalPackage(pkg PackageReference) (LocalPackageReference, error)
 	}
 	return LocalPackageReference{}, errors.Errorf("%q is not a local package", pkg.PackagePath())
 }
+
+func SortPackageReferencesByPathAndVersion(packages []PackageReference) {
+	sort.Slice(packages, func(i, j int) bool {
+		comparer := versionComparer{
+			left:  []rune(packages[i].PackagePath()),
+			right: []rune(packages[j].PackagePath()),
+		}
+		compare := comparer.Compare()
+		return compare < 0
+	})
+}
+
+// versionComparer captures our state while doing an alphanumeric version comparision
+// We need separate indexes for each side because we're doing a numeric comparison, which will
+// compare "100" and "0100" as equal (leading zeros are not significant)
+type versionComparer struct {
```

Comment:
really minor: Given how much of this file is now the `versionComparer` wondering if we should put it into its own file?

Reply:
It really does only apply to this type though so I see why it's here...